lib/local_daily_digest.js

```diff
@@ -0,0 +1,407 @@
+		},	
+		function (cb){
+			//Get a list of all checked out users in the last 24 hours for those locations
+			var i = 0;
+			async.forEachSeries(requiredLocationsId, function(location, callback2) {
+				Contact.find({ $query: { type: 'local', locationId: location, revised: {$gt: Date.now() - (24*60*60*1000)}, status: false},  $orderby : {location: -1} }, function (err, contact){
+						contact.forEach(function(contact){
+							if(!contact){
+								console.log("No users found1");
+								return cb();
+							}
+
+							if(tempCheckedOutUsers.length == 0){
+								tempCheckedOutUsers.push({
+									cOut_location: contact.location,
+									cOut_id: [contact._id],
+									cOut_profile: [contact._profile],
+									cOut_locationId: contact.locationId,
+									cOut_nameFamily:[contact.nameFamily],
+									cOut_nameGiven: [contact.nameGiven],
+									cOut_organization: [contact.organization],
+									cOut_bool: true,
+									globalProfile: []
+								});
+							}
+							else{
+								i = tempCheckedOutUsers.length -1;
+								if(tempCheckedOutUsers[i].cOut_location == contact.location){
+									tempCheckedOutUsers[i].cOut_id.push(contact._id);
+									tempCheckedOutUsers[i].cOut_profile.push(contact._profile);
+									tempCheckedOutUsers[i].cOut_nameGiven.push(contact.nameGiven);
+									tempCheckedOutUsers[i].cOut_nameFamily.push(contact.nameFamily);
+									tempCheckedOutUsers[i].cOut_organization.push(contact.organization[0].name);
+									tempCheckedOutUsers[i].globalProfile = [];
+								}
+								else{
+									tempCheckedOutUsers.push({
+										cOut_location: contact.location,
+										cOut_id: [contact._id],
+										cOut_profile: [contact._profile],
+										cOut_locationId: contact.locationId,
+										cOut_nameFamily:[contact.nameFamily],
+										cOut_nameGiven: [contact.nameGiven],
+										cOut_organization: [contact.organization],
+										cOut_bool: true,
+										globalProfile: []
+									});
+								}
+							}
+					});	
+				    callback2(); 
+				})
+			}, cb)
+		},
+		function(cb){
+				async.forEachSeries(tempCheckedOutUsers, function(user, callback3) {
+					async.forEachSeries(user.cOut_profile, function(id, callback2) {
+						Contact.findOne({ type: 'global', _profile: id}, function(err, contact){
+							globalProfileLinks.push(contact._id);
+							if(user.globalProfile.length != user.cOut_profile.length)
+								user.globalProfile.push(contact._id);
+
+						    callback2(); 
+						})
+					}, callback3)
+			}, cb);
+		},
+		function (cb){
+			/*
+			For each country a person is a manager/editor off,
+				For all the required contacts, get the 
+					location and locaiton id
+					profile id		
+					user name 
+					user organization data
+			*/
+			var checkInTemp = [], checkOutTemp = [];
+			allLocalProfilesWDigest.forEach(function(managerProfile){
+				managerProfile.dailyDigest.forEach(function(country){
+
+					checkedInUsers.forEach(function(user){
```

Comment:
You wouldn't need this loop if you were sorting by location right after the query